pkg/autodiscovery/providers/config_reader.go

```diff
@@ -325,77 +327,87 @@ func collectDir(parentPath string, folder os.FileInfo, integrationErrors map[str
 // GetIntegrationConfigFromFile returns an instance of integration.Config if `fpath` points to a valid config file
 func GetIntegrationConfigFromFile(name, fpath string) (integration.Config, error) {
 	cf := configFormat{}
-	config := integration.Config{Name: name}
+	conf := integration.Config{Name: name}
 
 	// Read file contents
 	// FIXME: ReadFile reads the entire file, possible security implications
 	yamlFile, err := readFilePtr(fpath)
 	if err != nil {
-		return config, err
+		return conf, err
 	}
 
 	// Parse configuration
 	// Try UnmarshalStrict first, so we can warn about duplicated keys
 	if strictErr := yaml.UnmarshalStrict(yamlFile, &cf); strictErr != nil {
 		if err := yaml.Unmarshal(yamlFile, &cf); err != nil {
-			return config, err
+			return conf, err
 		}
 		log.Warnf("reading config file %v: %v\n", fpath, strictErr)
 	}
 
 	// If no valid instances were found & this is neither a metrics file, nor a logs file
 	// this is not a valid configuration file
 	if cf.MetricConfig == nil && cf.LogsConfig == nil && len(cf.Instances) < 1 {
-		return config, errors.New("Configuration file contains no valid instances")
+		return conf, errors.New("Configuration file contains no valid instances")
 	}
 
 	// at this point the Yaml was already parsed, no need to check the error
 	if cf.InitConfig != nil {
 		rawInitConfig, _ := yaml.Marshal(cf.InitConfig)
-		config.InitConfig = rawInitConfig
+		conf.InitConfig = rawInitConfig
 	}
 
 	// Go through instances and return corresponding []byte
 	for _, instance := range cf.Instances {
 		// at this point the Yaml was already parsed, no need to check the error
 		rawConf, _ := yaml.Marshal(instance)
-		config.Instances = append(config.Instances, rawConf)
+		dataConf := (integration.Data)(rawConf)
+		if fargate.IsFargateInstance(context.TODO()) {
+			// In Fargate, since no host tags are applied in the backend,
+			// add the configured DD_TAGS/DD_EXTRA_TAGS to the instance tags.
+			tags := config.GetConfiguredTags(false)
```

Comment:
Same question. Is it because you also want this to support ECS?

Reply:
It's a little unclear to me at the moment whether it makes sense to add these EKS Fargate tags as well. Since there seems to be agreement here that we should, I'll change this.